### PR TITLE
Add Staff Assignment column to Project Enrollments and Households tables

### PR DESCRIPTION
**Pull request by martha**
## Description

[//]: # 'remove if not applicable'
GH issue: https://github.com/open-path/Green-River/issues/6336
Depends on hmis-warehouse PR: https://github.com/greenriver/hmis-warehouse/pull/4582

## Type of change
[//]: # 'remove options that are not relevant'
- [ ] Bug fix
- [x] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/modules/projects/components/tables/ProjectClientEnrollmentsTable.tsx [hunk 1]
```diff
@@ -146,6 +148,15 @@ export const ENROLLMENT_COLUMNS: {
       return null;
     },
   },
+  assignedStaff: {
+    ...ASSIGNED_STAFF_COL,
+    render: (e) => {
+      if ('household' in e && isRenderFunction(ASSIGNED_STAFF_COL.render)) {
```
Comment (ttoomey) [inline src/modules/projects/components/tables/ProjectClientEnrollmentsTable.tsx]: So it looks like isRenderFunction casts ASSIGNED_STAFF_COL.render to be `(e:any)=> React.ReactNode`. This seems like an indirect way of just casting e to `any`.

Suggest we either cast e in a more obvious way, or use a guard function. So:

```ts
    render: (e:any) => {
      if ('household' in e){
        return ASSIGNED_STAFF_COL.render(e.household);
      }
      return null;
    },
```

or


```ts
//guard function
const isHouseholdWithStaff = (e: any): e is {household: HouseholdWithStaffAssignmentsFragment} => {
  return 'household' in e && 'staffAssignments' in e.household
}
// ....

    render: (e) => {
      return isHouseholdWithStaff(e) ? ASSIGNED_STAFF_COL.render(e.household) : null
    }
```

File: src/modules/projects/components/tables/ProjectClientEnrollmentsTable.tsx [hunk 2]
```diff
@@ -178,20 +189,33 @@ const ProjectClientEnrollmentsTable = ({
...
   );
 
+  const {
+    project: { staffAssignmentsEnabled },
+  } = useProjectDashboardContext();
+
   const defaultColumns: ColumnDef<ProjectEnrollmentQueryEnrollmentFieldsFragment>[] =
     useMemo(() => {
-      return [
+      const c = [
```
Comment (ttoomey) [inline src/modules/projects/components/tables/ProjectClientEnrollmentsTable.tsx]: I know this is a small function but suggest we use more descriptive variable names, such as `cols` or just `results`

File: src/modules/projects/components/tables/ProjectHouseholdsTable.tsx [hunk 3]
```diff
@@ -32,6 +34,27 @@ const TableCellContainer = ({ children }: { children: ReactNode }) => (
...
 
+export const ASSIGNED_STAFF_COL = {
+  header: 'Assigned Staff',
+  optional: true,
+  defaultHidden: true,
+  render: (hh: HouseholdWithStaffAssignmentsFragment) => {
+    if (!hh?.staffAssignments || hh.staffAssignments.nodes.length === 0) return;
+
+    const first = hh.staffAssignments.nodes[0].user.name;
+    const rest = hh.staffAssignments?.nodesCount - 1;
```
Comment (ttoomey) [inline src/modules/projects/components/tables/ProjectHouseholdsTable.tsx]: micro optimization, I think this is a little more concise
```suggestion
    if (!hh.staffAssignments?.nodes.length) return;

    const first = hh.staffAssignments.nodes[0].user.name;
    const rest = hh.staffAssignments.nodesCount - 1;
```

**Review by ttoomey — APPROVED** [review]
looks good! I left a few optional suggestions

File: src/modules/projects/components/tables/ProjectClientEnrollmentsTable.tsx (hunk 1 above)
Comment (martha) [reply src/modules/projects/components/tables/ProjectClientEnrollmentsTable.tsx]: I like the idea of a guard function. Thanks for the suggestion @ttoomey !

**Pull request merged**